manifests/state-multus-cni/0040-configmap.yml

```diff
@@ -11,6 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+{{ if .CrSpec.Config -}}
```

Comment:
There is a general mechanims at state level where resources are being deleted if needed.
do you really think we need a finer grained mechanism ?

if user sets config to nil, its expected that the next steps for him is to re-create the config map with the given name.
so it will be deleted by the user.

thinking about it, NOT deleting it may be actually better, because then the user may know the name of the config map it needs to re-create :).

Reply:
> There is a general mechanims at state level where resources are being deleted if needed. do you really think we need a finer grained mechanism ?
This https://github.com/Mellanox/network-operator/blob/5fb589e9dcdecbca12867e941f7c5a00e62f94b5/pkg/state/state_multus_cni.go#L79 won't delete ConfigMap if Multus should be deployed

> if user sets config to nil, its expected that the next steps for him is to re-create the config map with the given name. so it will be deleted by the user.
> 
> thinking about it, NOT deleting it may be actually better, because then the user may know the name of the config map it needs to re-create :).

I don't understans this use-case when user deletes operator-generated ConfigMap and creates it's own. My assumption is that all configuration should be done via NicClusterPolicy

Reply:
> I don't understans this use-case when user deletes operator-generated ConfigMap and creates it's own. My assumption is that all configuration should be done via NicClusterPolicy

the use-case is having configuration provided out of band. this was an internal request.

Reply:
> the use-case is having configuration provided out of band. this was an internal request.

in this case, please add a note about this feature in a README and this will be good to merge

Reply:
> won't delete ConfigMap if Multus should be deployed
Yes, and i think we dont need a finer grained mechansim to handle it (for now...)

Reply:
added a section in README.